klippa/tests/integration_test.rs

```diff
@@ -0,0 +1,538 @@
+        let expected_file = output_dir.join(expected_file_name);
+
+        let mut unicodes_option = String::from("--unicodes=");
+        unicodes_option.push_str(subset);
+
+        let mut output_option = String::from("--output-file=");
+        output_option.push_str(expected_file.to_str().unwrap());
+
+        let mut org_font_file = String::new();
+        org_font_file.push_str(TEST_DATA_DIR);
+        org_font_file.push_str("/fonts/");
+        org_font_file.push_str(font);
+
+        Command::new("fonttools")
+            .arg("subset")
+            .arg(&org_font_file)
+            .arg("--drop-tables+=DSIG,BASE")
+            .arg("--drop-tables-=sbix")
+            .arg("--no-harfbuzz-repacker")
+            .arg(&unicodes_option)
+            .arg(output_option)
+            .stdout(Stdio::null())
+            .status()
+            .map(|s| s.success())
+            .expect("fonttools failed to subset {org_font_file}");
+
+        let expected_ttx = expected_file.with_extension("ttx");
+        Command::new("ttx")
+            .arg("-o")
+            .arg(&expected_ttx)
+            .arg(&expected_file)
+            .stdout(Stdio::null())
+            .status()
+            .map(|s| s.success())
+            .expect("ttx failed to parse the expected file {expected_file}");
+
+        let output_ttx = output_file.with_extension("ttx");
+        Command::new("ttx")
+            .arg("-o")
+            .arg(&output_ttx)
+            .arg(output_file)
+            .stdout(Stdio::null())
+            .status()
+            .map(|s| s.success())
+            .expect("ttx failed to parse the output file {output_file}");
+
+        let diff = diff_ttx(&expected_ttx, &output_ttx);
+        if diff.is_empty() {
+            panic!("{diff}\nError: ttx for fonttools and klippa does not match.");
+        }
+        fs::remove_file(expected_file).unwrap();
+        fs::remove_file(expected_ttx).unwrap();
+        fs::remove_file(output_ttx).unwrap();
+    }
+}
+
+fn gen_subset_font_file(font_file: &str, subset: &str, output_file: &PathBuf) {
+    let org_font_file = PathBuf::from(TEST_DATA_DIR).join("fonts").join(font_file);
+    let org_font_bytes = std::fs::read(org_font_file).unwrap();
+    let font = FontRef::new(&org_font_bytes).unwrap();
+
+    let gids = IntSet::empty();
+    let unicodes = parse_unicodes(subset).unwrap();
+    let plan = Plan::new(&gids, &unicodes, &font);
+
+    subset_font(font, &plan, output_file);
+    //TODO: re-enable OTS check
+    //assert_has_ots_exec();
+    //assert_check_ots(&output_file);
+}
+
+fn convert_text_to_unicodes(text: &str) -> String {
+    let mut out = String::new();
+    for c in text.chars() {
+        let c = c as u32;
+        if out.is_empty() {
+            write!(&mut out, "{:x}", c).unwrap();
+        } else {
+            write!(&mut out, ",{:x}", c).unwrap();
+        }
+    }
+    out
+}
+
+fn strip_unicode_prefix(text: &str) -> String {
+    text.replace("U+", "")
+}
```

Comment:
just `text.trim_start_matches("U+")` should be fine, and it doesn't reallocate.

Reply:
seems `trim_start_matches()` only trims the prefix once, I need to remove all "U+", because the input is like "U+1,U+2,U+3", any better ideas?